sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/UdfUdafRegister.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.extensions.sql;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.beam.sdk.transforms.Combine;
+import org.apache.beam.sdk.transforms.SerializableFunction;
+
+/** Register for UDF and UDAF. */
+public interface UdfUdafRegister {
+  default Map<String, Class<? extends BeamSqlUdf>> getEvalUdfs() {
```

Comment:
By reading this it's unclear what are `eval` udfs vs `serialize` udfs, add comments. And I would probably just call these methods `getBeamSqlUdfs()` and `getSerializableFunctionUdfs()` to avoid inventing another name for these things.

Another approach that I can think of (don't know if it's better or not) to make it a bit more extensible is to use `TypeDescriptor`-style type specification:

```
udfs = ImmutableMap.builder()
   .put(BeamSqlUdf.class, new HashMap<String, BeamSqlUdf>())
   .put(SerializableFunction.class, new HashMap<String, SerializableFunction<?, ?>>)
   .build();

public <T> Map<String, T> getUdf(TypeDescriptor<T> udfTypeDescriptor) {
   udfType = // get udf class from udfTypeDescriptor
   return udfs.get(udfType);
}

```

I imagine the usage will be like: `register.getUdfs(new TypeDescriptor<BeamSqlUdf>> {})`.

And I wonder wether it will be useful to have methods like `register.getUdf(String name, TypeDescriptor udfType);`